src/test/kotlin/ecommerce/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,364 @@
+                    INSERT INTO product (name, price, imageUrl) VALUES ('Superman', 1000, 'https://alexnsan.comics/imageurl/3');
+                    INSERT INTO product (name, price, imageUrl) VALUES ('Naruto', 1000, 'https://alexnsan.comics/imageurl/4');
+                    INSERT INTO product (name, price, imageUrl) VALUES ('Full Metal Alchemist', 1000, 'https://alexnsan.comics/imageurl/5');"""
+        jdbcTemplate.batchUpdate(query)
+
+        controller = ProductController(productService)
+    }
+
+    @Test
+    fun `create() - should insert product and return 201 when form is valid`() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(Product(name = "product1 [new]", price = 1.5, imageUrl = "https://www.product.com/image/1"))
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().log().all().extract()
+
+        Assertions.assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value())
+    }
+
+    @Test
+    fun `create() - should return 400 when name is blank`() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(Product(name = "", price = 1.5, imageUrl = "https://www.product.com/image/1"))
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().log().all().extract()
+
+        val targets =
+            listOf(
+                "Contains unallowed character",
+                "Product name is required",
+                "Must be no more than 15 characters, including spaces",
+            )
+        val resBody = response.jsonPath().getMap<String, String>("errors")
+        val actual = resBody["name"]
+        Assertions.assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value())
+        Assertions.assertThat(actual).isIn(targets)
+    }
+
+    @Test
+    fun `create() - should return 400 when name is more than 15 characters`() {
+        val response =
+            RestAssured
+                .given().log().all()
+                .body(
+                    Product(
+                        name = "this is very long name",
+                        price = 1.5,
+                        imageUrl = "https://www.product.com/image/1",
+                    ),
+                )
+                .contentType(ContentType.JSON)
+                .`when`().post("/api/products")
+                .then().log().all().extract()
+
+        val expected = "Must be no more than 15 characters, including spaces"
+        val resBody = response.jsonPath().getMap<String, String>("errors")
+        val actual = resBody["name"]
+        Assertions.assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value())
+        Assertions.assertThat(actual).isEqualTo(expected)
```

Comment:
```suggestion
        RestAssured
            .given().log().all()
            .body(
                Product(
                    name = "this is very long name",
                    price = 1.5,
                    imageUrl = "https://www.product.com/image/1",
                ),
            )
            .contentType(ContentType.JSON)
            .`when`().post("/api/products")
            .then().log().all()
            .assertThat()
            .statusCode(HttpStatus.BAD_REQUEST.value())
            .body("errors.name", equalTo("Must be no more than 15 characters, including spaces"))
```

This is not a must, but we can also let `RestAssured` assert everything :)

Reply:
OHHHHH! So cool! I will to that!